libfuzzer/libfuzz_harness.cc

```diff
@@ -42,26 +41,107 @@ int null_printf(FILE* stream, const char* format, ...)
     return 0;
 }
 
-/**
- * @brief Accept an input buffer and size.
- *
- * @param[in] data Pointer to the input buffer.
- * @param[in] size Size of the input buffer.
- * @return -1 if the input is invalid
- * @return 0 if the input is valid and processed.
- */
-int LLVMFuzzerTestOneInput(const uint8_t* data, std::size_t size)
+typedef std::unique_ptr<ubpf_vm, decltype(&ubpf_destroy)> ubpf_vm_ptr;
+
+ubpf_vm_ptr create_ubpf_vm(const std::vector<uint8_t>& program_code)
 {
-    // Assume the fuzzer input is as follows:
-    // 32-bit program length
-    // program byte
-    // test data
+    // Automatically free the VM when it goes out of scope.
+    std::unique_ptr<ubpf_vm, decltype(&ubpf_destroy)> vm(ubpf_create(), ubpf_destroy);
 
-    // Copy memory into a writable buffer.
-    std::vector<uint8_t> memory;
+    if (vm == nullptr) {
+        // Failed to create the VM.
+        // This is not interesting, as the fuzzer input is invalid.
+        // Do not add it to the corpus.
+        return vm;
+    }
+
+    ubpf_toggle_undefined_behavior_check(vm.get(), true);
```

Comment:
Is this a runtime check? Or a validation check? If the latter, do we also want to have this enabled in the jit path?

Reply:
I am not sure. Should we? I think the cost of doing bounds checks and undefined behavior checks are too high for the JIT path.